### PR TITLE
Added Ingress option w/ ssl for secure deploys

### PR DESCRIPTION
**Pull request by chainlink**
Moved configmap to values to allow for custom config map passing
Added initialdelay to allow pod to come up before health checks
Added dirty check to restart pods on configmap change
Updated notes with ingress docs

### CODE REVIEW

**Comment by unguiculus** [comment]
cc @etiennetremel

File: stable/verdaccio/templates/NOTES.txt [hunk 1]
```diff
@@ -1,5 +1,9 @@
 1. Get the application URL by running these commands:
-{{- if contains "NodePort" .Values.service.type }}
+{{- if .Values.ingress.enabled }}
+{{- range .Values.ingress.hosts }}
+  http://{{ . }}
```
Comment (unguiculus) [inline stable/verdaccio/templates/NOTES.txt]: Depending on `tls` config, should be `https`.

**Review by unguiculus — CHANGES_REQUESTED** [review]

**Comment by etiennetremel** [comment]
LGTM

File: stable/verdaccio/templates/NOTES.txt (hunk 1 above)
Comment (chainlink) [reply stable/verdaccio/templates/NOTES.txt]: happy to change this, but I can't find a good example on how to match the `ingress.hosts` array with the `ingress.tls[].hosts` to determine `http` vs `https`. Mind giving me a hand with the syntax?

File: stable/verdaccio/Chart.yaml [hunk 2]
```diff
@@ -1,7 +1,7 @@
 apiVersion: v1
 description: A lightweight private npm proxy registry (sinopia fork)
 name: verdaccio
-version: 0.1.3
+version: 0.2.0
 appVersion: 2.7.1
```
Comment (etiennetremel) [inline stable/verdaccio/Chart.yaml]: Would you mind updating to 2.7.3 (security fix) and change the default tag value in `stable/verdaccio/values.yaml`

File: stable/verdaccio/templates/NOTES.txt (hunk 1 above)
Comment (etiennetremel) [reply stable/verdaccio/templates/NOTES.txt]: not super esthetic but that work:
```
{{- $tls := .Values.ingress.tls }}
{{- if .Values.ingress.enabled }}
{{- range $host := .Values.ingress.hosts }}
{{- if $tls }}
https://{{ $host }}
{{- else }}
http://{{ $host }}
{{- end }}
{{- end }}
```

**Review by etiennetremel — CHANGES_REQUESTED** [review]

**Comment by chainlink** [comment]
@unguiculus @etiennetremel addressed your comments

**Comment by etiennetremel** [comment]
LGTM

**Review by unguiculus — APPROVED** [review]

**Comment by chainlink** [comment]
Looks like they haven't pushed 2.7.3 yet, I created an issue in the repo https://github.com/verdaccio/verdaccio/issues/489

**Comment by chainlink** [comment]
2.7.3 has been added! https://hub.docker.com/r/verdaccio/verdaccio/tags/

**Comment by chainlink** [comment]
Can you please re-run tests?

**Comment by chainlink** [comment]
g2g!

**Comment by chainlink** [comment]
Can we get this merged in pls?

**Review by etiennetremel — APPROVED** [review]

File: stable/verdaccio/templates/ingress.yaml [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    {{- range $key, $value := .Values.ingress.annotations }}
+      {{ $key }}: {{ $value | quote }}
+    {{- end }}
+spec:
+  rules:
+    {{- range $host := .Values.ingress.hosts }}
+    - host: {{ $host }}
+      http:
+        paths:
+          - path: /
```
Comment (JohnLindahlTech) [inline stable/verdaccio/templates/ingress.yaml:23]: @chainlink: path: `/` does not automatically add a wildcard match in GCE so this ingress will only exactly match `/`. Removing the path entirely help in my specific case but atleast I would also need `/*` to serve anything but the root html.

Comment (chainlink) [reply stable/verdaccio/templates/ingress.yaml:23]: Should we default to` /*` then?

Comment (JohnLindahlTech) [reply stable/verdaccio/templates/ingress.yaml:23]: I would certainly like that. I hope and think that Nginx-ingresses wont be affected in a negative way.

Comment (MichielDeMey) [reply stable/verdaccio/templates/ingress.yaml:23]: Just hit this issue as well, using `/` will only route the root and not all other child routes need to serve the UI and more. Changing it to `/*` will work.
Just created a PR for this: https://github.com/kubernetes/charts/pull/6726

**Comment by MichielDeMey** [comment]
I believe that as a requirement for an ingress controller, the service must be of type `NodePort`. Currently it's set to `ClusterIP`.
Something we could easily fix in the template based on the `ingress.enabled` configuration value.

**Comment by chainlink** [comment]
I think that's only a requirement for the GLB L7 ingress (not the nginx ingress)

**Comment by MichielDeMey** [comment]
Ah, apologies. I've only ever used the GLB ingress. 😅

**Comment by chainlink** [comment]
No prob! (I've only ever used Nginx ingress) hence the discrepancies here. I wonder how we could pick up on that in the chart? 🤔

**Comment by MichielDeMey** [comment]
I don't believe we can, I checked the output of `helm create` and it defaults to a `ClusterIP` value in the `values.yaml` file. So I believe it's fine, since you can configure it. 👍

**Pull request merged**